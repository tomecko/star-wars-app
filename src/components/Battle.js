import React from 'react';
import { Badge, Card, CardGroup } from 'reactstrap';

export function Battle({ battle, resourceInfo }) {
  const { items, winners } = battle;
  return (
    <CardGroup>
      {items && items.map((item, i) => (
        <Card
          key={i}
          body
          className="text-center"
          inverse={winners.includes(i) ? true : undefined}
          color={winners.includes(i) ? 'primary' : undefined}
        >
          <p>
            {item.name}<br/>
            <Badge>
              {resourceInfo.scoreDescription}: {resourceInfo.getScoreText(item)}
            </Badge>
          </p>
        </Card>
      ))}
    </CardGroup>
  );
}

