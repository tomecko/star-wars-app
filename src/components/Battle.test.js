import React from 'react';
import renderer from 'react-test-renderer';

import { Battle } from './Battle';

test('matches snapshot', () => {
  const battle = {
    items: [
      { name: 'n1', a: 1 },
    ],
    winners: [1],
  };
  const resourceInfo = {
    name: 'resourceName',
    getScore: ({ a }) => a,
    getScoreText: ({ a }) => String(a),
    scoreDescription: 'a',
  }
  const component = renderer.create(
    <Battle battle={battle} resourceInfo={resourceInfo} />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
