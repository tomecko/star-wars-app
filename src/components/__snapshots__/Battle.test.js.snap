// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="card-group"
>
  <div
    className="text-center card card-body"
  >
    <p>
      <strong>
        n1
      </strong>
      <br />
      <span
        className="badge badge-secondary"
      >
        a
        : 
        1
      </span>
    </p>
  </div>
</div>
`;
